crates/router/src/connector/stripe.rs

```diff
@@ -1026,23 +1027,70 @@ impl api::IncomingWebhook for Stripe {
             .parse_struct("StripeWebhookObjectId")
             .change_context(errors::ConnectorError::WebhookReferenceIdNotFound)?;
 
-        Ok(api_models::webhooks::ObjectReferenceId::PaymentId(
-            api_models::payments::PaymentIdType::ConnectorTransactionId(details.data.object.id),
-        ))
+        Ok(match details.data.object.object.as_str() {
+            "payment_intent" => api_models::webhooks::ObjectReferenceId::PaymentId(
+                api_models::payments::PaymentIdType::ConnectorTransactionId(details.data.object.id),
+            ),
+            "dispute" => api_models::webhooks::ObjectReferenceId::PaymentId(
+                api_models::payments::PaymentIdType::ConnectorTransactionId(
+                    details
+                        .data
+                        .object
+                        .payment_intent
+                        .ok_or(errors::ConnectorError::WebhookReferenceIdNotFound)?,
+                ),
+            ),
+            _ => Err(errors::ConnectorError::FailedToObtainAuthType)?,
```

Comment:
Correct the error type
```suggestion
            _ => Err(errors::ConnectorError::WebhookReferenceIdNotFound)?,
```